### DISCUSSION TITLE
Path setting in GDL DE

### DISCUSSION THREAD

**Discussion by rdimeo** [gnudatalanguage/gdl#1518] [Q&A]
Folks - I am relatively new to the Eclipse DE for GDL. Is there a simple way (i.e. a preferences setting in the DE) to set the path variable so that includes all the directories and subdirectories so I don't have to force a compile for each of my library routines? I found it straightforward in the IDL DE but I'm struggling a bit to find a comparable dialog in the GDL DE. My current workaround is to compile and run a .PRO file that modifies the !PATH environment variable accordingly when I start my GDL session in the DE.

Thanks in advance!

Rob

**Comment by alaingdl** [comment]
I am not sure to understand what you try to do, but we have a PATH_ADD procedure in src/pro/utilities (path_add.pro)
with a /recursive keyword.

You may also use an export in .bashrc or equivalent file : export GDL_PATH='+/path/to/gdl/'

**Reply by rdimeo** [reply]
The PATH_ADD does what my workaround does. Every time I launch the DE, I have to compile and run my short .PRO file that adds the paths to all directories that contain my library files. I was looking for a way for the GDL DE to automatically add paths to my library of routines every time I launch it. I don't know how to do that.

**Comment by GillesDuvert** [comment]
I see that the IDL DE automatically adds the project directory and all its subdirectories to the !PATH when a project is opened, and remove them when it is closed. Obviously the GDL DE is not so clever. Yet.  Help welcome? At least, this should be reported as an issue to the GDLDE project.

In the meantime, for the record, one can
- either define it in  GDL_PATH (or IDL_PATH) environment variable, or 
- put a path-changing command in the startup file (defined in the env. var GDL_STARTUP (or IDL_STARTUP for compatibility).

But this will be done only once at start of the GDL DE so it does not answer your problem.
